deps/reopt_vcg/src/ReoptVCG/ABI.lean

```diff
@@ -54,71 +112,146 @@ namespace LLVMType
 --   | _ => False
 -- | _ => False
 
+-- FIXME: gets around lean bug/missing feature with inductive datatypes containing others.
+@[reducible]
+def prim_toSmtSort? : LLVMType -> Option SmtSort 
+| prim pt => pt.toSmtSort?
+| vector n (prim pt) => 
+  (NTuple.make ∘ List.replicate n) <$> pt.toSmtSort?
+| _       => none
+
 @[reducible]
-def HasBVRepr : LLVMType -> Prop
-| LLVM.LLVMType.ptr _  => True
-| LLVM.LLVMType.prim pt  => PrimType.HasBVRepr pt
-| LLVM.LLVMType.vector _ (LLVM.LLVMType.prim pt) => PrimType.HasBVRepr pt
-| _ => False
+def toSmtSort? : LLVMType -> Option SmtSort
+| ptr _   => some (bitvec 64)
+| prim pt => pt.toSmtSort?
+| vector n ty => 
+  if n = 0 
+  then none -- FIXME
```

Comment:
`panic! none` here?

Reply:
Actually, if it just returns `none` and we get an error that's fine - I guess `panic!` would only be helpful if it led to a "silent failure"